pkg/code-generator/codegen/templates/snapshot_emitter_template.go

```diff
@@ -233,9 +250,189 @@ func (c *{{ lower_camel .GoName }}Emitter) Snapshots(watchNamespaces []string, o
 			}
 		}(namespace)
 	}
+	if hasWatchedNamespaces && opts.ExpressionSelector != "" {
+		// watch resources using non-watched namespaces. With these namespaces we
+		// will watch only those that are filted using the label selectors defined
+		// by Expression Selectors
+
+		// first get the renaiming namespaces
+		var k kubernetes.Interface
+		excludeNamespacesFieldDesciptors := ""
+
+		var buffer bytes.Buffer
+		for i, ns := range watchNamespaces {
+			buffer.WriteString("metadata.namespace!=")
+			buffer.WriteString(ns)
+			if i < len(watchNamespaces)-1 {
+				buffer.WriteByte(',')
+			}
+		}
+		excludeNamespacesFieldDesciptors = buffer.String()
```

Comment:
any limits in field selector length we should be concerned about?

Reply:
Don't think this is relevant anymore.